src/mds/MDSRank.cc

```diff
@@ -743,10 +743,6 @@ void MDSRankDispatcher::tick()
   // update average session uptime
   sessionmap.update_average_session_age();
 
-  if (is_active() || is_stopping()) {
-    mdlog->trim();  // NOT during recovery!
-  }
-
```

Comment:
I think we can remove `mdlog->trim()` for shutdown below this too, yes?

I also think we can make `MDLog::trim` a private method. Consequently, we could get rid of the `upkeep_mutex` and only use `submit_mutex`. The upkeep thread will acquire/drop the `submit_mutex` and call into the private `trim` method which takes a `std::unique_lock` argument (so it can drop it for `try_to_expire`).